R/utils.R

```diff
@@ -381,3 +384,31 @@ build_datanames_error_message <- function(label = NULL,
     }
   )
 }
+
+disable_teal_tabs <- function(ns, hide_content = TRUE) {
+  shinyjs::disable(selector = sprintf(".teal-body:has('#%s') .nav li a", ns("options_buttons")))
+  if (hide_content) {
+    shinyjs::runjs(
+      sprintf(
+        '$("#%s .tab-content").hide()',
+        ns("tabpanel_wrapper")
+      )
+    )
+  }
+}
+
+enable_teal_tabs <- function(ns) {
+  shinyjs::enable(selector = sprintf(".teal-body:has('#%s') .nav li a", ns("options_buttons")))
+  shinyjs::runjs(
+    sprintf(
+      '$("#%s .tab-content").show()',
+      ns("tabpanel_wrapper")
+    )
+  )
+  shinyjs::runjs(
+    sprintf(
+      '$("#%s .tab-content").trigger("shown");',
+      ns("tabpanel_wrapper")
+    )
+  )
```

Comment:
I think nothing. `.trigger("shown")` is the `shiny` method to retrigger calculation. It is useful for elements which have been hidden and their `renderXYZ` have been disabled because of being in the hidden state. Normally `shinyjs::hide` and `shinyjs::show` do their job. But if you're hiding html-elements containing shiny outputs with pure javascript code, then you need `.trigger("shown")`. Is that right @vedhav ?